app/main.py

```diff
@@ -201,30 +224,70 @@ def update_live_alerts_data(alert):
 
     It returns :
 
-    - a json containing live_alerts data where is_acknowledged is False.
+    - a json containing live alerts data, where live alerts either correspond to a not-yet-acknowledged event or have
+    been created less than 12 hours ago;
+
     - a dict where keys are event id and value lists of all urls related to the same event.
 
-    These url come from the API calls, triggered by the websocket message stroed in msg hidden div.
+    These URLs come from the API calls, triggered by the websocket message stored in msg hidden div.
     """
 
     # Fetching live alerts where is_acknowledged is False
     response = api_client.get_ongoing_alerts().json()
-    all_alerts = pd.DataFrame(response)
-    if all_alerts.empty:
+
+    # If there is no alert, we prevent the callback from updating anything
+    if len(response) == 0:
         raise PreventUpdate
-    live_alerts = all_alerts.loc[~all_alerts["is_acknowledged"]]
+
+    # We store all alerts in a DataFrame and we want to select "live alerts",
+    # Ie. alerts that either correspond to a not-yet-acknowledged event or have been created less than 12 hours ago
+    all_alerts = pd.DataFrame(response)
+
+    # We first want to build the boolean indexing mask that corresponds to the time-related condition
+    # We convert values in the "created_at" column to datetime format
+    all_alerts['created_at'] = pd.to_datetime(all_alerts['created_at'], utc=True)
+
+    # For each of these creation dates, we check whether they were registered less than 12 hours ago
+    # This provides us with the first boolean indexing mask
+    mask_time = all_alerts['created_at'].map(
+        lambda x: pd.Timestamp(datetime.now(tz=pytz.UTC)) - x) <= pd.Timedelta('12 hours')
+
+    # We now want to build the boolean indexing mask that indicates whether or not the event is unacknowledged
+    # We start by making an API call to fetch all events
+    url = cfg.API_URL + '/events/'
+    all_events = requests.get(url, headers=api_client.headers).json()
+
+    # Then, we construct a dictionary whose keys are the event IDs (as integers) and values are the corresponding
+    # "is_acknowledged" field in the events table (boolean)
+    is_event_acknowledged = {}
+    for event in all_events:
+        is_event_acknowledged[event['id']] = event['is_acknowledged']
+
+    # We map this dictionary upon the column and revert the booleans with ~ as we want unacknowledged events
+    mask_acknowledgement = ~all_alerts['event_id'].map(is_event_acknowledged)
+
+    # We link the two masks with an OR condition
+    mask = np.logical_or(mask_time, mask_acknowledgement)
```

Comment:
Shouldn't we want an AND condition here ? Meaning that live_alerts are those referring an unacknowledged event lasting for more that 12 hours ?

Reply:
With an AND, an alert/event would disappear from the platform as soon as it is acknowledged (or to be more precise, at the next update of the alert data), which we wanted to avoid by adding a time condition. So, I do believe that this should be an OR, with three possibilities for an alert to be displayed: 

- Either it is still "fresh" as it was created less than 12 hours ago;
- Or it has not yet been acknowledged by firemen (how can we be sure that they taken it into account otherwise?);
- Or both. 

Let me know if this remains unclear 😊